workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs

```diff
@@ -30,20 +53,48 @@ public void SetProjectName(string projectName)
 
         public void SetLoginToken(string loginToken)
         {
+            if (loginToken.Equals(string.Empty) && LocatorParameters.CredentialsType.Equals(null))
+            {
+                return;
+            }
+
             LocatorParameters.CredentialsType = LocatorCredentialsType.LoginToken;
             LocatorParameters.LoginToken = new LoginTokenCredentials
             {
                 Token = loginToken
             };
         }
 
+        public void SetSteamCredentials(string steamToken, string deploymentTag)
+        {
+            if (steamToken.Equals(string.Empty))
+            {
+                return;
+            }
+
+            LocatorParameters.CredentialsType = LocatorCredentialsType.Steam;
+            LocatorParameters.Steam = new SteamCredentials
+            {
+                Ticket = steamToken,
+                DeploymentTag = deploymentTag
+            };
+        }
+
         public static LocatorConfig CreateConnectionConfigFromCommandLine(Dictionary<string, string> parsedArgs)
         {
             var config = new LocatorConfig();
             var loginToken = CommandLineUtility.GetCommandLineValue(
                 parsedArgs, RuntimeConfigNames.LoginToken, string.Empty);
             var projectName = CommandLineUtility.GetCommandLineValue(
                 parsedArgs, RuntimeConfigNames.ProjectName, string.Empty);
+            var steamDeploymentTag = CommandLineUtility.GetCommandLineValue(
+                parsedArgs, RuntimeConfigNames.SteamDeploymentTag, string.Empty);
+            var steamToken = CommandLineUtility.GetCommandLineValue(
+                parsedArgs, RuntimeConfigNames.SteamToken, string.Empty);
+
+            // Assign Steam Credentials first so that loginToken can fall back to default values if not provided
+            config.SetSteamCredentials(steamToken, steamDeploymentTag);
```

Comment:
a login token will fail as well, if it uses the default values. We literally need to provide a login token or steam credentials to go successfully through the locator flow.  So we should have a check here: do we have a steamToken? yes, go set it, no, go set the logintoken

Reply:
Having a second look at the implementation I have now baked the logic in the functions to cover the cases you mentioned.